src/rgen/Microsoft.Macios.Generator/Emitters/BindingSyntaxFactory.Dlfcn.cs

```diff
@@ -0,0 +1,447 @@
+				IdentifierName ("dlsym").WithTrailingTrivia (Space)
+			)
+		).WithArgumentList (ArgumentList (SeparatedList<ArgumentSyntax> (arguments)).NormalizeWhitespace ());
+
+		// *((TYPE *) dlsymCall)
+		var castExpression = PrefixUnaryExpression (
+			SyntaxKind.PointerIndirectionExpression,
+			ParenthesizedExpression (
+				CastExpression (
+					PointerType (
+						IdentifierName (blittableType)),
+					dlsymInvocation.WithLeadingTrivia (Space)
+				)));
+
+		var compilationUnit = CompilationUnit ().WithMembers (
+			SingletonList<MemberDeclarationSyntax> (
+				GlobalStatement (
+					ExpressionStatement (castExpression
+					))));
+		return compilationUnit;
+	}
+
+
+	public static CompilationUnitSyntax SetBlittableField (string nsObjectType, string libraryName, string fieldName,
+		string value)
+		=> throw new NotImplementedException ();
+
+	/// <summary>
+	/// Returns the getter needed to access the native value exposed by a field property. The method will return the
+	/// Dflcn calls needed.
+	/// </summary>
+	/// <param name="property">A field property under code generation.</param>
+	/// <returns>The appropriate Dlfcn call to retrieve the native value of a field property.</returns>
+	/// <exception cref="NotSupportedException">When the caller tries to generate the call for a no field property.</exception>
+	/// <exception cref="NotImplementedException">When the property type is not supported for a field property.</exception>
+	public static CompilationUnitSyntax FieldConstantGetter (in Property property)
+	{
+		// check if this is a field, if it is not, we have an issue with the code generator
+		if (!property.IsField)
+			throw new NotSupportedException ("Cannot retrieve getter for non field property.");
+
+		// retrieve all the necessary data from the info field of the property
+		var libraryName = property.ExportFieldData.Value.LibraryName;
+		var symbolName = property.ExportFieldData.Value.FieldData.SymbolName;
+
+		if (property.ReturnType.IsNSObject) {
+			return property.ReturnType.Name == "Foundation.NSString"
+				? GetStringConstant (libraryName, symbolName)
+				// all nsobjects are retrieved using the same generic getter
+				: GetNSObjectField (property.ReturnType.Name, libraryName, symbolName);
+		}
+
+		// keep the formatting to make it more readable
+#pragma warning disable format
+		// use the return type and the special type of the property to decide what getter we are going to us
+		return property.ReturnType switch {
+			// special types
+			{ Name: "CoreGraphics.CGSize" } => GetCGSize (libraryName, symbolName),
+			{ Name: "CoreMedia.CMTag" } => GetStruct ("CoreMedia.CMTag", libraryName, symbolName),
+			{ Name: "nfloat" } => GetNFloat (libraryName, symbolName),
+			{ Name: "System.Drawing.SizeF" } => GetSizeF (libraryName, symbolName),
+
+			// Billable types 
+			{ Name: "CoreMedia.CMTime" or "AVFoundation.AVCaptureWhiteBalanceGains" }
+				=> GetBlittableField (property.ReturnType.Name, libraryName, symbolName),
```

Comment:
This could just use Dlfcn.GetStruct. In fact all blittable structs could be used with GetStruct, there's no need to have an approved list of structs.